src/index.tsx

```diff
@@ -72,12 +72,17 @@ const formatOptions = ({
       ? ['welcome', 'auth', ...mandatorySteps]
       : ['welcome', ...mandatorySteps]
 
+  //@ts-ignore TODO: quick fix, remove this whole block when the welcome screen is configured via workflow
+  const welcomeStep: StepConfig[] = [
+    steps?.map(formatStep).find((i) => i.type === 'welcome'),
+  ] || [{ type: 'welcome' }]
+
   return {
     ...otherOptions,
     smsNumberCountryCode: validateSmsCountryCode(smsNumberCountryCode),
     useWorkflow,
     steps: useWorkflow
-      ? defaultSteps.map(formatStep)
+      ? (welcomeStep || defaultSteps).map(formatStep)
```

Comment:
If `defaultSteps` is suppose to be a fallback, then this doesn't work as welcome step always returns an array (`[undefined]`).

Reply:
@DannyvanderJagt very good point, I actually thought I had a condition to default the welcome step, update it in the block above, not so clean. what do you think?

Reply:
Not sure what the logic here should be?

This will always use `welcomeStep` (as it's always populated by either the welcome step from `steps` or `[{ type: 'welcome' }]` and always ignore `defaultSteps`. For auth this will therefore remove the step `auth`.

If we don't use auth with workflow, then best to remove the `defaultSteps` from line 85 to clean it up. 

If you just want to update the welcome step if it's available in `Onfido.init` then something like this should work:

```tsx
 // @ts-ignore TODO: quick fix, remove this whole block when the welcome screen is configured via workflow
  const welcomeStep =
    steps?.map(formatStep).find((i) => i.type === 'welcome') || 'welcome'

  const defaultSteps: (StepTypes | StepConfig)[] =
    process.env.SDK_ENV === 'Auth'
      ? [welcomeStep, 'auth', ...mandatorySteps]
      : [welcomeStep, ...mandatorySteps]

  return {
    ...otherOptions,
    smsNumberCountryCode: validateSmsCountryCode(smsNumberCountryCode),
    useWorkflow,
    steps: useWorkflow
      ? defaultSteps.map(formatStep)
      : (steps || defaultSteps)
          .map(formatStep)
          .map(injectPrivateConfig)
          .filter(({ type }) => !internalSteps.includes(type)),
  }
```

Reply:
@DannyvanderJagt gotcha!!!! I totally discounted Auth ( in reality I think it is not part of workflow anyway) , in my head the polling would start after welcome screen ( at the moment welcome screen mandatory for workflow) so we always start with welcome screen. 
Thank you very much for your brilliant feedback.